contracts/ATM/IATMGovernance.sol

```diff
@@ -0,0 +1,240 @@
+        @param dataTypeIndex index of this data type.
+        @param dataProviderAddress address of the added Data Provider.
+     */
+    event DataProviderAdded(
+        address indexed signer,
+        uint8 indexed dataTypeIndex,
+        address dataProviderAddress
+    );
+
+    /**
+        @notice Emitted when a Data Provider was updated on this ATM.
+        @param signer transaction sender address.
+        @param dataTypeIndex index of this data type.
+        @param oldAddress previous address of the Data Provider.
+        @param newAddress new address of the Data Provider.
+     */
+    event DataProviderUpdated(
+        address indexed signer,
+        uint8 indexed dataTypeIndex,
+        address oldAddress,
+        address newAddress
+    );
+
+    /**
+        @notice Emitted when a Data Provider was removed on this ATM.
+        @param signer transaction sender address.
+        @param dataTypeIndex index of this data type.
+        @param dataProviderAddress address of the Data Provider.
+     */
+    event DataProviderRemoved(
+        address indexed signer,
+        uint8 indexed dataTypeIndex,
+        address dataProviderAddress
+    );
+
+    /**
+        @notice Emitted when a new CRA - Credit Risk Algorithm is set.
+        @param signer transaction sender address.
+        @param craCommitHash github commit hash with the new CRA implementation.
+     */
+    event CRASet(address indexed signer, string indexed craCommitHash);
+
+    /* External Functions */
+
+    /**
+        @notice Adds a new General Setting to this ATM.
+        @param settingName name of the setting to be added.
+        @param settingValue value of the setting to be added.
+     */
+    function addGeneralSetting(bytes32 settingName, uint256 settingValue) external;
+
+    /**
+        @notice Updates an existing General Setting on this ATM.
+        @param settingName name of the setting to be modified.
+        @param newValue new value to be set for this settingName. 
+     */
+    function updateGeneralSetting(bytes32 settingName, uint256 newValue) external;
+
+
+    /**
+        @notice Removes a General Setting from this ATM.
+        @param settingName name of the setting to be removed.
+     */
+    function removeGeneralSetting(bytes32 settingName) external;
+
+    /**
+        @notice Adds a new Asset Setting from a specific Market on this ATM.
+        @param asset market specific asset address.
+        @param settingName name of the setting to be added.
+        @param settingValue value of the setting to be added.
+     */
+    function addAssetMarketSetting(address asset, bytes32 settingName, uint256 settingValue) external;
+
+    // /**
+    //     @notice Updates an existing Asset Setting from a specific Market on this ATM.
+    //     @param asset market specific asset address.
+    //     @param settingName name of the setting to be added.
+    //     @param settingValue value of the setting to be added.
```

Comment:
Please, remove the commented code.

Reply:
Oh that's to avoid compilation errors but I wanted to share with you the interface as it will be. I am removing comments as I am implementing methods.